sdk/sdk-core/aar/src/main/java/mil/emp3/core/mapgridlines/AbstractMapGridLine.java

```diff
@@ -222,31 +226,84 @@ public void run() {
                 double viewWidthInMeters;
                 Point westPoint;
                 Point eastPoint;
-                IGeoPosition centerWest = new GeoPosition();
-                IGeoPosition centerEast = new GeoPosition();
-
-                centerWest.setLatitude(AbstractMapGridLine.this.boundingBox.centerLatitude());
-                centerWest.setLongitude(AbstractMapGridLine.this.boundingBox.getWest());
+                IGeoPosition centerWest;
+                IGeoPosition centerEast;
+                Point northPoint;
+                Point southPoint;
+                IGeoPosition centerNorth;
+                IGeoPosition centerSouth;
+                ICamera camera = AbstractMapGridLine.this.currentCamera;
+                IMapInstance mapInstance = AbstractMapGridLine.this.mapInstance;
+                IEmpBoundingBox bBox = AbstractMapGridLine.this.boundingBox;
+
+                if ((Math.abs(camera.getHeading()) <= 5.0) &&
```

Comment:
Move this code (the if block that has a special case calculation for bounding box) to BoundingBoxGeneration class and call it a method (calculateFor AllCornersTouchedSpecialCase that returns true or false. Then cal it on line 183 (just like the call on line 180).With this approach everyone gets the same bounding Box and all calculations are in one place. All those 5.0 make them constants in the class aand explain what ithey are used for and they are experimental.